includes/parserhooks/SetParserFunction.php

```diff
@@ -41,6 +46,15 @@ public function __construct( ParserData $parserData, MessageFormatter $messageFo
 	}
 
 	/**
+	 * @since  2.2
+	 *
+	 * @param Parser $parser
+	 */
+	public function setParser( $parser ) {
```

Comment:
Somehow it should be made very clear to the users of this class that `setParser` needs to be called before `parse`, since else you can end up with a fatal